apps/frontend/src/components/cards/controltable/ControlRowHeader.vue

```diff
@@ -71,6 +71,11 @@
     <template #id>
       <v-card-text class="pa-2 title font-weight-bold">
         {{ control.data.id }}
+        {{
+          control.data.tags['legacy']
+            ? '(' + control.data.tags['legacy'][0] + ')'
```

Comment:
Can't assume that legacy will be an array of strings.  The types say that control.data.tags is a mapping between keys that are strings and anything.  Consequently, we have to explicitly typecheck that the value is either a string or an array whose first value is a string.  For anything else or if the legacy tag doesn't exist in the first place, then don't display anything.  If someone complains, we can loosen this restriction at that time.

Reply:
Good catch. So, check if an array, then check if the first or only value is a string, and if not convert it internally.